remediation-service/handler/evaluation_done_handler.go

```diff
@@ -11,95 +12,89 @@ import (
 	"os"
 )
 
-// EvaluationDoneEventHandler handles incoming evaluation-done events
-type EvaluationDoneEventHandler struct {
+// EvaluationFinishedEventHandler handles incoming evaluation-done events
+type EvaluationFinishedEventHandler struct {
 	KeptnHandler *keptnv2.Keptn
 	Event        cloudevents.Event
 	Remediation  *Remediation
 }
 
 // HandleEvent handles the event
-func (eh *EvaluationDoneEventHandler) HandleEvent() error {
-	evaluationDoneEventData := &keptn.EvaluationDoneEventData{}
+func (eh *EvaluationFinishedEventHandler) HandleEvent() error {
+	evaluationDoneEventData := &keptnv2.EvaluationFinishedEventData{}
 
 	err := eh.Event.DataAs(evaluationDoneEventData)
 	if err != nil {
 		eh.KeptnHandler.Logger.Error("Could not parse evaluation-done event: " + err.Error())
 		return err
 	}
 
-	if evaluationDoneEventData.TestStrategy != "real-user" {
-		eh.KeptnHandler.Logger.Info("Ignoring evaluation-done event with testStrategy " + evaluationDoneEventData.TestStrategy)
+	remediations, err := getRemediationsByContext(eh.KeptnHandler.KeptnContext, eh.KeptnHandler.Event)
+	if err != nil {
+		eh.KeptnHandler.Logger.Error(fmt.Sprintf("could not retrieve open remediations for keptnContext %s: %s", eh.KeptnHandler.KeptnContext, err.Error()))
+		return err
+	}
+
+	if len(remediations) == 0 {
+		eh.KeptnHandler.Logger.Info(fmt.Sprintf("No open remediations for keptnContext %s", eh.KeptnHandler.KeptnContext))
 		return nil
 	}
 
 	if evaluationDoneEventData.Result == "pass" || evaluationDoneEventData.Result == "warning" {
-		msg := "Remediation successful. Remediation actions resulted in evaluation result: " + evaluationDoneEventData.Result
+		msg := "Remediation successful. Remediation actions resulted in evaluation result: " + string(evaluationDoneEventData.Result)
 		eh.KeptnHandler.Logger.Info(msg)
 		return eh.Remediation.sendRemediationFinishedEvent(keptn.RemediationStatusSucceeded, keptn.RemediationResultPass, msg)
 	}
 
 	// get remediation.yaml
 	resource, err := eh.Remediation.getRemediationFile()
 	if err != nil {
+		_ = eh.Remediation.sendRemediationFinishedEvent(keptn.RemediationStatusErrored, keptn.RemediationResultFailed, err.Error())
```

Comment:
Can we also log the error here, as done in line 59: `eh.KeptnHandler.Logger.Error(err.Error())` - to tell the user that the `remediation.yaml` was not found.

Reply:
good catch - has been updated